src/mod/irc.mod/chan.c

```diff
@@ -333,7 +325,7 @@ static int detect_chan_flood(char *floodnick, char *floodhost, char *from,
     chan->floodwho[which][0] = 0;
     if (which == FLOOD_DEOP)
       chan->deopd[0] = 0;
-    u = get_user_from_member(m);
+    u = lookup_user_record(m, NULL, from);
```

Comment:
missing the account name we might have, m is always true for everything but FLOOD_JOIN so I'd try to make that clear in the code:

```suggestion
    if (which == FLOOD_JOIN) {
      u = lookup_user_record(m, victim_or_account, from);
    } else {
      u = get_user_from_member(m);
    }
```